R/read_file.R

```diff
@@ -0,0 +1,80 @@
+#' Read files with extension .fst, .csv, and .rda (.rds and .RData)
+#'
+#' @param file_name Character; path to a file storing a single object.
+#'
+#' @noRd
+#' @examples
+#' # Odd case: .RData saved with extension .rda
+#' path <- tempfile(fileext = ".rda")
+#'
+#' data1 <- data.frame(x = 1)
+#' saveRDS(data1, path)
+#' read_file(path)
+#'
+#' # More natural case: .RData saved as .rda
+#' data2 <- data.frame(x = 1)
+#' save(data2, file = path)
+#' read_file(path)
+#'
+#' # Fails by design: .RData with more than 1 object
+#'
+#' save(data1, data2, file = path)
```

Comment:
I believe the first argument of `save()` should be a list of objects, rather than `...`

Reply:
From `?load()` and this experiment, I think both are equivalent:

``` r
data1 <- data2 <- data.frame(x = 1)

path1 <- tempfile(fileext = ".rda")
save(data1, data2, file = path1)

path2 <- tempfile(fileext = ".rda")
save(list = c("data1", "data2"), file = path2)

e1 <- new.env()
base::load(path1, envir = e1)

e2 <- new.env()
base::load(path2, envir = e2)

identical(ls(e1), ls(e2))
#> [1] TRUE
```

<sup>Created on 2020-10-07 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Reply:
oops....

from the help file...
```
save(..., list = character(),
```

```
...  the names of the objects to be saved (as symbols or character strings).
list  A character vector containing the names of objects to be saved.
```